modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/RaftGroupServiceImpl.java

```diff
@@ -446,11 +523,37 @@ private boolean recoverable(Throwable t) {
      * @return Random node.
      */
     private Peer randomNode() {
-        List<Peer> peers0 = peers;
+        List<PeerId> peers0 = peers;
 
         if (peers0 == null || peers0.isEmpty())
             return null;
 
-        return peers0.get(current().nextInt(peers0.size()));
+        return peerFromPeerId(peers0.get(current().nextInt(peers0.size())));
+    }
+
+    private Peer peerFromPeerId(PeerId peer) {
+        if (peer == null)
+            return null;
+        else
+            return new Peer(NetworkAddress.from(peer.getEndpoint().getIp() + ":" + peer.getEndpoint().getPort()));
+    }
+
+    private List<PeerId> parsePeerList(List<String> peers) {
+        List<PeerId> res = new ArrayList<>(peers.size());
+        for (String peer: peers) {
```

Comment:
Empty lines are missing before for and beofre return.
{} for one liner are excessive.